firmware/arduino/chessboard/uart.ino

```diff
@@ -23,70 +52,79 @@ void serialEvent2()
         }
 
         // Check if the buffer is full, process input
-        if (byteNum  ==  5)
+        if (byteNum  ==  6)
         {
             // Reset buffer position
             byteNum = -1;
 
             // Process input
             // Returns true for valid input and false for invalid input. Calls movement function
-            parseMessageFromPi(buffer);
+            currentState = EXECUTING;
+            if (parseMessageFromPi(buffer))
+                errorCode = NONE;
+
+            // Send response
+            sendMessageToPi(currentState, buffer[5], errorCode);
         }
     }
 }
 
 bool parseMessageFromPi(char * message)
 {
-    // Move types 0 and 3 are valid for the same range of values
-    if (message[0]  ==  0  ||  message[0]  ==  3)
+    // Move types 0 and 1 are valid for the same range of values
+    if (message[0]  ==  DIRECT  ||  message[0]  ==  EDGES)
     {
         // Check that the inputs are in a valid range
-        if (message[1] < 'a'  ||  message[1] > 'h' 
-           ||  message[2] < '1'  ||  message[2] > '8' 
-           ||  message[3] < 'a'  ||  message[3] > 'h' 
-           ||  message[4] < '1'  ||   message[4] > '8') 
+        if (message[1] < 'A'  ||  message[1] > 'L' 
```

Comment:
Since you do this so often, consider making a function `isValidCoord(char c)` to do this check for you.
```
def isInvalidCoord(char c) {
    return c < 'A || c > 'L';
```